This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sun, 29 Dec 2019 11:20:00 +0100
Subject: [PATCH 1/2] Fix streamiu signal parameter types

The parameters of the read and seek signal were flipped.

diff --git a/libvips/iofuncs/streamiu.c b/libvips/iofuncs/streamiu.c
index 1111111..2222222 100644
--- a/libvips/iofuncs/streamiu.c
+++ b/libvips/iofuncs/streamiu.c
@@ -182,9 +182,9 @@ vips_streamiu_class_init( VipsStreamiuClass *class )
 		G_SIGNAL_ACTION,
 		G_STRUCT_OFFSET( VipsStreamiuClass, read ), 
 		NULL, NULL,
-		vips_INT64__INT64_INT,
+		vips_INT64__POINTER_INT64,
 		G_TYPE_INT64, 2,
-		G_TYPE_INT64, G_TYPE_INT );
+		G_TYPE_POINTER, G_TYPE_INT64 );
 
 	/**
 	 * VipsStreamiu::seek:
@@ -204,9 +204,9 @@ vips_streamiu_class_init( VipsStreamiuClass *class )
 		G_SIGNAL_ACTION,
 		G_STRUCT_OFFSET( VipsStreamiuClass, seek ), 
 		NULL, NULL,
-		vips_INT64__POINTER_INT64,
+		vips_INT64__INT64_INT,
 		G_TYPE_INT64, 2,
-		G_TYPE_POINTER, G_TYPE_INT64 );
+		G_TYPE_INT64, G_TYPE_INT );
 
 }
 
From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sun, 29 Dec 2019 11:40:00 +0100
Subject: [PATCH 2/2] Ensure size_t is casted to gint64

size_t could be 32-bit.

diff --git a/libvips/iofuncs/streamiu.c b/libvips/iofuncs/streamiu.c
index 1111111..2222222 100644
--- a/libvips/iofuncs/streamiu.c
+++ b/libvips/iofuncs/streamiu.c
@@ -83,7 +83,7 @@ vips_streamiu_read_real( VipsStreami *streami,
 	bytes_read = 0;
 
 	g_signal_emit( streami, vips_streamiu_signals[SIG_READ], 0,
-		buffer, length, &bytes_read );
+		buffer, (gint64) length, &bytes_read );
 
 	VIPS_DEBUG_MSG( "  vips_streamiu_read_real, seen %zd bytes\n", 
 		bytes_read );
diff --git a/libvips/iofuncs/streamou.c b/libvips/iofuncs/streamou.c
index 1111111..2222222 100644
--- a/libvips/iofuncs/streamou.c
+++ b/libvips/iofuncs/streamou.c
@@ -83,7 +83,7 @@ vips_streamou_write_real( VipsStreamo *streamo,
 	bytes_written = 0;
 
 	g_signal_emit( streamo, vips_streamou_signals[SIG_WRITE], 0,
-		data, length, &bytes_written );
+		data, (gint64) length, &bytes_written );
 
 	VIPS_DEBUG_MSG( "  %zd\n", bytes_written );
 
